packages/core/src/components/button/buttons.tsx

```diff
@@ -73,13 +87,27 @@ export const AnchorButtonFactory = React.createFactory(AnchorButton);
 function getButtonClasses(props: IButtonProps) {
     return classNames(
         Classes.BUTTON,
-        { [Classes.DISABLED]: props.disabled },
+        { [Classes.DISABLED]: isButtonDisabled(props) },
         Classes.iconClass(props.iconName),
         Classes.intentClass(props.intent),
         props.className,
     );
 }
 
+function isButtonDisabled(props: IButtonProps) {
+    return props.disabled === true || props.isLoading === true;
+}
+
+function buttonContentsVisibility(isLoading: boolean) {
+    return {visibility: isLoading ? "hidden" : "visible"};
+}
+
+function maybeRenderSpinner(isLoading: boolean) {
+    return isLoading != null && isLoading
+      ? <span className="pt-button-loading-spinner"><Spinner className="pt-small" /></span>
```

Comment:
I'm not sure we can. I want the parent here to take up the full area of the button so I can center the spinner using `flex`. It may be possible without the extra element... @giladgray @adidahiya ?